rules.go

```diff
@@ -191,6 +191,8 @@ func gocriticWrapperFunc(m fluent.Matcher) {
 func gocriticNilValReturn(m fluent.Matcher) {
 	m.Match(`if $*_; $v == nil { return $v }`).
```

Comment:
Should we drop this one? or combine them into one? @quasilyte

Reply:
I thought that this pattern catches both if statements with and without init clause. I'll re-check this with gogrep.

Reply:
Buuuuuut we can make two rules with suggestion. Not one with some info and one with a fix.

Reply:
That's correct. But the form with the fix should go **before** this one.
Otherwise, the more generic pattern will match and we'll never get a suggestion.
At least it looks like we need to move the new rule above this one.